### DISCUSSION TITLE
Can I turn off Validation step when overfit_batches=X?

### DISCUSSION THREAD

**Discussion by IgorHoholko** [Lightning-AI/pytorch-lightning#11057] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I call `trainer.fit(model=model_lit, datamodule=datamodule)`

I have set parameters:

```
limit_val_batches: 0
limit_test_batches: 0
```

`datamodule.val_dataloader()` returns **None**

But it is still trying to perform validation... 
Is it a problem of lightning, or am I doing something wrong? :)

![image](https://user-images.githubusercontent.com/27057946/145991475-291a9199-22f8-43ce-a195-127d992ab09e.png)

**Comment by IgorHoholko** [comment]
Well. The reason why validation was called is `overfit_batches=1` parameter.

**Reply by IgorHoholko** [reply]
Why ?

![image](https://user-images.githubusercontent.com/27057946/145993051-27e128b5-42e7-4c8e-9316-0003cddf0794.png)

**Comment by rohitgr7** [comment, marked as answer]
it's now turned off by default on master:
https://github.com/PyTorchLightning/pytorch-lightning/blob/2faaf35b91a00aff397609a875a66c87f8ed6390/pytorch_lightning/trainer/trainer.py#L636-L640

**Reply by IgorHoholko** [reply]
great, thanks :)
Good time to update packages
